src/components/Notifications/Notifier.js

```diff
@@ -5,36 +5,40 @@ import { withSnackbar } from 'notistack';
 import { removeSnackbar } from '../../actions/notificationActions';
 
 class Notifier extends Component {
-    state = {
-        displayed: [],
-    };
+    displayed = [];
 
-    storeDisplayed = (key) => {
-        this.setState(({ displayed }) => ({
-            displayed: [...displayed, key],
-        }));
+    storeDisplayed = (id) => {
+        this.displayed = [...this.displayed, id];
     };
 
-    render() {
-        const { notifications, enqueueSnackbar, removeSnackbar } = this.props;
-        const { displayed } = this.state;
+    shouldComponentUpdate({ notifications: newSnacks = [] }) {
+        const { notifications: currentSnacks } = this.props;
+        let notExists = false;
+        for (let i = 0; i < newSnacks.length; i += 1) {
+            if (notExists) continue;
+            notExists = notExists || !currentSnacks.filter(({ key }) => newSnacks[i].key === key).length;
```

Comment:
Looks like a candidate for optimization, returning as soon as the condition evaluates to true, and returning false outside the loop.

Would reduce the number of iterations and remove one variable

EDIT: Just noticed the `continue` - makes me feel that this refactor would be guaranteed to be less smelly code! 😄

Reply:
This code came straight from the package docs, as I am not sure how the system works internally, maybe it is intended to loop through all the snacks

Reply:
The `continue` makes me think otherwise, though! I'd vote for changing it and seeing if it breaks something 🛠

Reply:
I refactored it into 
```js
for (let i = 0; i < newSnacks.length; i += 1) {
   if (!currentSnacks.filter(({ key }) => newSnacks[i].key === key).length) 
        return true;
}
return false;
```

and the notifications seem to be working normally, I even checked the redux store and it also seems fine, so I think this was a nice catch

Reply:
Nice! Better code is always cool!
Well, but not the best there is... The best idea is just to have [nocode](https://github.com/kelseyhightower/nocode). 😉